controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -29,21 +42,155 @@ type RegisteredClusterReconciler struct {
 	APIExtensionClient apiextensionsclient.Interface
 	Log                logr.Logger
 	Scheme             *runtime.Scheme
+	MceCluster         []helpers.MceInstance
 }
 
 func (r *RegisteredClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = context.Background()
 	logger := r.Log.WithValues("namespace", req.Namespace, "name", req.Name)
 	logger.Info("Reconciling...")
 
+	instance := &singaporev1alpha1.RegisteredCluster{}
+
+	if err := r.Client.Get(
+		context.TODO(),
+		types.NamespacedName{Namespace: req.Namespace, Name: req.Name},
+		instance,
+	); err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, giterrors.WithStack(err)
+	}
+
+	// create managecluster on creation of registeredcluster CR
+	if err := r.createManagedCluster(req.Name, ctx); err != nil {
+		logger.Error(err, "failed to create ManagedCluster")
+
+		return ctrl.Result{}, err
+	}
+
+	// update status of registeredcluster - add import command
+	if err := r.updateImportCommand(instance, req.Name, req.Namespace, ctx); err != nil {
+		logger.Error(err, "failed to update import command")
+
+		return ctrl.Result{}, err
+	}
+
 	return ctrl.Result{}, nil
 }
 
+func (r *RegisteredClusterReconciler) updateImportCommand(regCluster *singaporev1alpha1.RegisteredCluster, name string, namespace string, ctx context.Context) error {
+	// get import secret from mce managecluster namespace
+
+	importSecret := &corev1.Secret{}
+	if err := r.MceCluster[0].APIReader.Get(ctx, types.NamespacedName{Namespace: name, Name: name + "-import"}, importSecret); err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
What happens if the import secret isn't ready yet? Seems like we'd need to  be sure to re-reconcile until it is. The other option would be to add a Watcher for the import secret. Thoughts on which would be better?

Reply:
Do we know that the import secret is created with all of its content ready to go?

Reply:
if import secret is not found then it will reconcile again. Or I can add requeue after 30sec to reconcile again. yes when import secret is created, it is ready to apply its content on the managedcluster.

Reply:
If we put watch on secret resource, it can lead to pod in OOMKilled(memory issue) as no of secrets on cluster are more. 
If we want to use watch on secret, we need separate controller with customize cache to only cache those secrets. For that I think we need to put label on import-secret to only cache those secrets based on label.

Reply:
It looks like it currently won't reconcile if the secret is not found because if errors.IsNotFound(err) will be true.. Right? I don't know details on what is involved to generate the import secret so not sure what requeue time makes sense. I think 30 seconds will be too long. Maybe just one second? The user will be watching the UI for the import command.